lib/ui/page/home/page/my_profile/controller.dart

```diff
@@ -194,41 +197,43 @@ class MyProfileController extends GetxController {
       }
     }
 
-    listInitIndex = router.profileSection.value?.index ?? 0;
-
-    bool ignoreWorker = false;
-    bool ignorePositions = false;
+    observerController.initialIndex = router.profileSection.value?.index ?? 0;
 
     _profileWorker = ever(
       router.profileSection,
       (ProfileTab? tab) async {
-        if (ignoreWorker) {
-          ignoreWorker = false;
-        } else {
-          ignorePositions = true;
-          await itemScrollController.scrollTo(
-            index: tab?.index ?? 0,
-            duration: 200.milliseconds,
-            curve: Curves.ease,
-          );
-          Future.delayed(Duration.zero, () => ignorePositions = false);
+        if (tab == _lastSelectedProfileTab) return;
 
-          highlight(tab);
-        }
+        _lastSelectedProfileTab = tab;
+        _isIgnoreObserving = true;
+
+        await observerController.animateTo(
+          index: tab?.index ?? 0,
+          duration: 200.milliseconds,
+          curve: Curves.ease,
+        );
+
+        highlight(tab);
       },
     );
 
-    positionsListener.itemPositions.addListener(() {
-      if (!ignorePositions) {
-        final ProfileTab tab = ProfileTab
-            .values[positionsListener.itemPositions.value.first.index];
-        if (router.profileSection.value != tab) {
-          ignoreWorker = true;
-          router.profileSection.value = tab;
-          Future.delayed(Duration.zero, () => ignoreWorker = false);
-        }
+    onObserve = (resultModel) {
```

Comment:
Also I can see there being some problems with determining the correct active profile section in the menu tab when scrolling the page up and down. On the video you can see `Login options` section being considered active when it shouldn't, etc. 

https://github.com/user-attachments/assets/3459566b-284d-4a1a-a2ca-21a41406638f

Reply:
That's new behavior of detecting current tab for visible elements. Before this we were detecting the first visible element and now it's the last element that is visible for at least 90%, if none of visible elements satisfies that condition then last element at view is being selected as current tab. This new behavior allows for correct detection of all the tabs through scrolling from the beginning to the end and every tab item is being selected through process of scrolling.

But this change pulled some other problems, that you have pointed in other conversations. So i think it would be better to just return  old behavior, but it'll also have old problems with not all tabs selected on scroll, and automatic changing of tab when jumping to the end of the view, to the earlier tabs.

Should I try to correct a little that new behavior or just return an old one? What do you think?

Reply:
@McKoog, you should prefer the implementation that lacks any bugs.

In the approach you mentioned there's also a heuristic that the user is currently looking at the bottom block, and that's why you select the bottom one as the active. However this might not be true, because application doesn't track the eye movement of the user. They might look at the top block, or at the bottom block. Thus I suggest not trying to implement any fancy mechanisms there, as they won't be smart enough to really display the currently active block.

Reply:
fixed on 3bdb4c364a9c19145ac3970c4389cdc034f8f30f commit